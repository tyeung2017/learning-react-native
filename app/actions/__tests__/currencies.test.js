import { getInitialConversion, GET_INITIAL_CONVERSION, changeCurrencyAmount, CHANGE_CURRENCY_AMOUNT } from '../currencies';

it('test getInitialCopnversion', () => {
  const expected = {
    type: GET_INITIAL_CONVERSION,
  };
  const actual = getInitialConversion();

  expect(actual).toEqual(expected);
});

it('test changeCurrencyAmount', () => {
  const expected = {
    type: CHANGE_CURRENCY_AMOUNT,
    amount: 1000,
  };

  const actual = changeCurrencyAmount(1000);

  expect(actual).toEqual(expected);
});

it('snapshot of test getInitialCopnversion', () => {
  expect(getInitialConversion()).toMatchSnapshot();
});

it('snapshot of test changeCurrencyAmount', () => {
  expect(changeCurrencyAmount(1000)).toMatchSnapshot();
  expect(changeCurrencyAmount('1000')).toMatchSnapshot();// for testing string input
});

