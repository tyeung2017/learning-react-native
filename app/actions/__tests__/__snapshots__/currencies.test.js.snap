// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of test changeCurrencyAmount 1`] = `
Object {
  "amount": 1000,
  "type": "CHANGE_CURRENCY_AMOUNT",
}
`;

exports[`snapshot of test changeCurrencyAmount 2`] = `
Object {
  "amount": 1000,
  "type": "CHANGE_CURRENCY_AMOUNT",
}
`;

exports[`snapshot of test getInitialCopnversion 1`] = `
Object {
  "type": "GET_INITIAL_CONVERSION",
}
`;
